src/shared/components/case-editor/services/cases.service.ts

```diff
@@ -290,4 +294,53 @@ export class CasesService {
     });
   }
 
+  private processTasksOnSuccess(caseData: any, eventData: any, profile: Profile): void {
+    const user = profile && profile.user && profile.user.idam ? profile.user.idam.id : null;
+    /**
+     * Retrieve all tasks, which are:
+     *   * For the current case;
+     *   * Can be completed (status is not "Completed" or "Cancelled");
+     *   * Can be completed by the current event;
+     *   * Are assigned to the current user.
+     */
+    const taskSearchParameter: any = {
+      ccdId: caseData.id,
+      eventId: eventData.id,
+      state: [ 'Open' ] // Need to know which are the "completeable" statuses.
+    };
+    if (user) {
+      taskSearchParameter.user = [ user ];
+    }
+
+    // Now get the tasks using those properties.
+    this.getWorkAllocationTasks([ taskSearchParameter ]).subscribe((tasks: any[]) => {
+      console.log('Got tasks', tasks);
+      if (tasks && tasks.length > 0) {
+        if (tasks.length === 1) {
+          // TODO: Attempt to mark this task as complete.
+          // If this fails, flag it as an error.
+        } else {
+          // TODO: This is a problem. We need to flag it up as an error.
+        }
+      }
+      // If we didn't get any tasks back, we're all good. Nothing to see here.
+    });
+  }
+
+  public getWorkAllocationTasks(taskSearchParameters: any[]): Observable<object> {
```

Comment:
This is kinda the main bit that needs looking at. Is this in the right place? Have I understood the requirement correctly? Is the order in which it's processed correct (as in, have I called it from the right place)?